plugins/modules/folder_template.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Ansible Cloud Team (@ansible-collections)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: folder_template
+short_description: Create a template in a local VCenter folder from an existing VM
+description:
+- >-
+  This module creates a template in a local VCenter folder from an existing VM. The folder must already exist.
+  The VM must be powered off, and is otherwise unchanged.
+author:
+- Ansible Cloud Team (@ansible-collections)
+requirements:
+- pyvmomi
+options:
+  vm_name:
+    description:
+      - The name of the vm to be used to create the template
+    type: str
+    required: True
+  folder:
+    description:
+      - The name of the folder that the new template should be placed in
+    type: str
+    required: True
+  name:
```

Comment:
I think template name will be more clear 
and for modification - 
new_template_name (not sure if it's supported)